File: src/main/java/dev/christopherbell/libs/common/api/contracts/Message.java
package dev.christopherbell.libs.common.api.contracts;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Represents a potential error or any information that will be return to the caller.
 */
@AllArgsConstructor
@Builder
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@NoArgsConstructor
public class Message {

  @JsonProperty("code")
  private String code;
  @JsonProperty("description")
  private String description;
}

File: common-api/src/main/kotlin/Message.kt
package com.christopherbell.dev.libs

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.io.Serializable

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
class Message : Serializable {
    @JsonProperty("code")
    private val code: String? = null

    @JsonProperty("description")
    private val description: String? = null
}